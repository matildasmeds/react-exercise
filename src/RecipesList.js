import React, { useState, useEffect } from 'react';
import Recipe from './Recipe.js';
import { H1 } from './BasicComponents.js';
import axios from 'axios';
import Urls from './Urls.js'

function RecipesList() {
  const [recipes, setRecipes] = useState([]);

  useEffect(() => {
    axios(Urls.recipes.list).then(result => {
      setRecipes(result.data);
    });
  }, []);

  console.log(recipes);

  return <div className="pa4 black-80">
      <H1 label='All recipes' />
      { recipes && recipes.map(recipe =>
        <div key={recipe.id}>
          <Recipe
            name={recipe.name}
            description={recipe.description}
            ingredients={recipe.ingredients}
          />
         </div>)}
    </div>;
};

export default RecipesList;
