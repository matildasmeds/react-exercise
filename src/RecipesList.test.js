import '@testing-library/jest-dom/extend-expect';

import React from 'react';
import { render, fireEvent, waitForElement } from '@testing-library/react';
import RecipesList from './RecipesList.js';

test('renders the list correctly', () => {
  const recipes = [{
    id: 1,
    name: 'My favorite recipe',
    description: 'Easy, healthy, cheap, delicious',
    ingredients: [{
      name: 'Secret ingredient'
    }]
  }];
  const { getByText } = render(<RecipesList recipes={recipes} />);
  [
    'My favorite recipe',
    'Easy, healthy, cheap, delicious',
    'Secret ingredient'
  ].forEach((str) => {
    const html = getByText(str);
    expect(html).not.toBeNull();
    expect(html).toMatchSnapshot();
  });
 });
