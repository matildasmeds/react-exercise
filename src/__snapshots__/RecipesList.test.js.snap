// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the list correctly 1`] = `
<h3
  class="f4 fwl baskerville mt0 lh-title"
>
  My favorite recipe
</h3>
`;

exports[`renders the list correctly 2`] = `
<p>
  Easy, healthy, cheap, delicious
</p>
`;

exports[`renders the list correctly 3`] = `
<div>
  Secret ingredient
</div>
`;
